### ISSUE TITLE
"Talking to {Towner} is not available in the shareware version" text is split over multiple lines

### ISSUE THREAD

**Issue by ephphatha** [diasurgical/devilutionX#2564]
The way text is drawn for conversation/store dialog in stores.cpp (StartTalk, around line 1208) means that it should be possible to render this text as a single string (letting the later call to DrawString on line 2270 wrap as needed) or manually split the string to 4 lines after loading the translation and substituting the relevant name.

This would allow translating the entire sentence in a way that is more natural for languages with different sentence structure. Also makes the line spacing more robust as the current approach doesn't handle cases where one line ends up significantly longer than the others.

**Comment by julealgon** [comment]
Isn't this a more general issue with how text is rendered that affects multiple places?

**Comment by AJenbo** [comment]
This also affects Wirt's dialog in the full version of the game. The way shop menus works makes this hard to fix. The easiest might be to wrap the line, and then split it in to multiple lines before injecting each line individually in to the menu, or completely rewrite how menus are rendered and work.

**Comment by ephphatha** [comment]
Good point about this affecting multiple places, I was approaching this from the perspective of having a nicer translation context (since I was looking through the translation template at the time).

![devilutionx screenshot showing the dialog menu when talking to an npc with overflowing text](https://user-images.githubusercontent.com/357684/129015042-32ce6e86-92a3-405f-a6d6-40e4ec6973cd.png)

While testing it looks like the draw routine isn't wrapping properly either... all four lines are meant to be centered so the first line of text should've wrapped more times than it actually did and wrapped earlier in the string.

Given translations are converted to latin1 if the source text is UTF8 the stopgap of splitting the string into `n` approximately equal lines after loading the translated value should work. Probably best to wait for #2505 to be merged before making that change.

**Comment by AJenbo** [comment]
> translations are converted to latin1 if the source text is UTF8

This hopefully won't hold true for very long as I'm planning to switch to using UTF8 internally in the engine.

**Comment by romanstingler** [comment]
<img width="768" height="665" alt="Image" src="https://github.com/user-attachments/assets/efbfa0bc-43a4-4c42-80bc-5d599bdf24ff" />
@AJenbo I think this is resolved
